Added support for customized mc agent spawn location

snyxan opened the pull request:
# Description

As title says, now you can pass agent spawn location as an argument when start the agent.

Try something like: 

`python agents/craftassist/craftassist_agent.py --no_default_behavior --agent_debug_mode --spawn_location 5 70 9`


## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [ ] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [ ] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

**Before**

Agent will always be spawned at a given location

**After**

You can specify where to spawn the agent

# Testing

Tested locally and passed.

# Checklist:

- [ ] I have performed manual end-to-end testing of the feature in my environment.
- [ ] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added relevant collaborators to review the PR before merge.
- [ ] [Polymetis only] I ran on hardware (1) all scripts in `tests/scripts`, (2) asv benchmarks.

kavyasrinet commented:
That and I believe we need to remake the CAgent too right ?
What does `agent_debug_mode` enable ?

snyxan commented:
> That and I believe we need to remake the CAgent too right ? 

That's right, I updated the instructions to include this step, thanks!

> What does `agent_debug_mode` enable ?

It's a mode where the agent fails gracefully instead of throwing errors and hangs. It's not related to this PR so I removed it to avoid confusion.

kavyasrinet reviewed (approved):
Whoops I reviewed the code a long time back and for some reason I thought I had approved this
